### ISSUE TITLE
Swiping away the left hand menu bar should work (?)

### ISSUE THREAD

**Issue by voltagex** [nolanlawson/pokedex.org#11]
I tried swiping away the left menu bar instead of hitting the button. I'm used to this for Android apps but I'm not sure if it's possible to do in a web app.

**Comment by nolanlawson** [comment]
Yeah, this is actually one of the few things that you can't currently do in a webapp, at 60fps anyway - having an animation that follows the user's thumb as they drag it along. Hence all of my animations are FLIP - you tap, the animation starts, the animation ends, and there's no pause/rewind or following user input.

I think maybe the Web Animation API is capable of solving it, but I'm not 100% sure. @paullewis / @rachelnabors would know!

**Comment by paullewis** [comment]
I feel like this should be possible by a) debouncing, b) using rAF + transforms. There may be input latency, but the actual motion should be 60fps. I'm in work tomorrow so I can take a look and see if it's even feasible. If not, I can see a bug being filed in my near future.

**Comment by nolanlawson** [comment]
@paullewis Wow, if that is actually possible to pull off at 60fps, I would be super impressed. I have never seen a side menu that can actually do this. Do you think it would be possible to progressively darken the main view as well, as many Android native apps do?

**Comment by paullewis** [comment]
Here's how it looks on [Voice Memos](https://voice-memos.appspot.com) (on a Nexus 5) when I drag-swipe the side nav away:

![screen shot 2015-11-27 at 9 53 09 am](https://cloud.githubusercontent.com/assets/617438/11438436/c5b63356-94ec-11e5-94c1-51c6c3068462.png)

Code of interest is here: https://github.com/GoogleChrome/voice-memos/blob/master/src/scripts/controller/AppController.js#L60-L93

I'm banking on the fact that we only get one touchmove event per frame here, though in general I advise coalescing input events, and dealing with them in a single rAF callback. But YMMV.

Also the code could be extended to dynamically alter the opacity value; I didn't, but it's compositor-friendly to do that (assuming you have applied a 3D transform or `will-change: opacity`)

**Comment by nolanlawson** [comment]
Holy wow, that is the best side-swiping menu in a webapp I've ever seen. It's 60fps in Chrome on my Nexus 5x, and nearly 60fps in Firefox as well. Amazing work.

Is it possible to also swipe it open when the menu is *hidden*? In the current implementation, it looks like you have to tap the hamburger, then you can drag the menu.

**Comment by mayankchd** [comment]
Yes it is possible to swipe it in . I'm trying to create a demo

 code-labs.io from google developers has a nice swipe in and out nav drawer which runs at 60fps .Check it :+1:  https://www.code-labs.io/codelabs/chrome-es2015/index.html

**Comment by jakearchibald** [comment]
Implementation note: swiping from the background area should work too. I reckon it's a small change to Paul's example.

**Comment by nickcolley** [comment]
Note on some devices those gestures are tied to OS level behaviours like the back button or previous history etc. Not sure the best way to avoid that as you can't query those behaviours with a web api?

**Comment by jgw96** [comment]
The polymer side menu has swiping in and out. Haven't looked at the code for it though. @paullewis, that's pretty amazing work
